packages/rum-core/src/common/observers/page-visibility.js

```diff
@@ -76,9 +76,26 @@ export function observePageVisibility(configService, transactionService) {
  * @param transactionService
  */
 function onPageHidden(configService, transactionService) {
-  reportInp(transactionService)
+  const inpTr = reportInp(transactionService)
+  // we don't want to flush the queue for every transaction that is ended when page becomes hidden
+  // and given the async nature of the ending process of transactions
```

Comment:
nit: might reword it as transaction ends are scheduled async as microtasks(promise), so we are coordinating the flushing process